freeipa-client/src/main/java/com/sequenceiq/freeipa/client/operation/AbstractFreeipaOperation.java

```diff
@@ -0,0 +1,56 @@
+package com.sequenceiq.freeipa.client.operation;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import com.google.common.collect.Maps;
+import com.sequenceiq.freeipa.client.FreeIpaClient;
+import com.sequenceiq.freeipa.client.FreeIpaClientException;
+import com.sequenceiq.freeipa.client.model.RPCResponse;
+
+public abstract class AbstractFreeipaOperation<R> {
+
+    public abstract String getMethodName();
+
+    public abstract Optional<R> invoke(FreeIpaClient freeipaClient) throws FreeIpaClientException;
+
+    public void appendOperationToBatchCall(List<Object> operations) {
```

Comment:
this is a bit odd for me. I would expect that this could return the result map, and the caller would add the Map to the list/collection